src/main/java/com/github/u3games/eventengine/security/DualBoxProtection.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2015-2016 L2J EventEngine
+ *
+ * This file is part of L2J EventEngine.
+ *
+ * L2J EventEngine is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * L2J EventEngine is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.github.u3games.eventengine.security;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.github.u3games.eventengine.datatables.ConfigData;
+import com.l2jserver.gameserver.network.L2GameClient;
+
+/**
+ * DualBox Protection based on WAN IP Address and tracert obtained directly from the client.
+ * @author Sacrifice
+ */
+public final class DualBoxProtection
+{
+	private Map<IpPack, Integer> _address = new HashMap<DualBoxProtection.IpPack, Integer>();
+	
+	public DualBoxProtection()
+	{
+		
+	}
+	
+	public synchronized boolean registerConnection(L2GameClient client)
+	{
+		if (ConfigData.DUALBOX_PROTECTION_ENABLED)
+		{
+			IpPack pack = new IpPack(client.getConnection().getInetAddress().getHostAddress(), client.getTrace());
+			Integer count = _address.get(pack);
+			if (count == null)
+			{
+				_address.put(pack, count = 1);
+			}
+			if (count < ConfigData.DUALBOX_MAX_ALLOWED)

```

Comment:
So, if the count is null, it will be 1. And when enters in this block, it will add another 1, so the count will be 2 for first client.

I suggest this:

```java
Integer count = _address.get(pack) == null ? 0 : _address.get(pack);
if (count < ConfigData.DUALBOX_MAX_ALLOWED)
{
  _address.put(pack, count =+ 1);
  return true;
}
return false;
```